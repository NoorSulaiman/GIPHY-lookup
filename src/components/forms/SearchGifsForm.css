#formField,#search{
    width:100%;
}

@media only screen and (max-width:786px){
    #searchButton{
        margin-top:2%;
    }
}