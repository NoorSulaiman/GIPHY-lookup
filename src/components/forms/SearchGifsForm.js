import React from 'react';
import PropTypes from 'prop-types';
import { Form, Button } from 'semantic-ui-react';
import InlineError from '../messages/InlineError';
import './SearchGifsForm.css'

class SearchGifsForm extends React.Component {
    state = {
        data: { search: '' },
        loading: false,
        errors: {},
    };

    onChange = e =>
        this.setState({
            ...this.state,
            data: { ...this.state.data, [e.target.name]: e.target.value },
        });

    onSubmit = e => {
        e.preventDefault();
        const errors = this.validate(this.state.data);
        this.setState({ errors });
        if (Object.keys(errors).length === 0) {
            this.setState({ loading: true });
            this.props.submit(this.state.data.search).then(this.setState({
                loading: false,
            }));
        }
    };

    validate = data => {
        const errors = {};
        if (data.search.length === 0 || !data.search.trim()) {
            errors.search = "Can't be blank";
        }
        return errors;
    };

    render() {
        const { data, errors, loading } = this.state;
        return (
            <Form id='searchForm' onSubmit={this.onSubmit} loading={loading}>
                <Form.Group inline>
                    <Form.Field id='formField' error={!!errors.search}>
                        <input
                            type="text"
                            id="search"
                            name="search"
                            placeholder="Search for GIFs"
                            value={data.search}
                            onChange={this.onChange}
                        />
                    </Form.Field>
                    <Button id='searchButton'>Search</Button>
                </Form.Group>
                {errors.search && <InlineError text={errors.search} />}
            </Form>
        );
    }
};
SearchGifsForm.propTypes = {
    submit: PropTypes.func.isRequired,
};

export default SearchGifsForm;