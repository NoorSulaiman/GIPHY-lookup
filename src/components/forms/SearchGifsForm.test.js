import React from 'react';
import { shallow } from 'enzyme';
import SearchGifsForm from './SearchGifsForm';

const clickFn = jest.fn();
describe('SearchGifsForm', () => {
    it('should render input form correctly with given function', () => {
        const component = shallow(<SearchGifsForm submit={clickFn} />);
        expect(component).toMatchSnapshot();
    });
    it('should fail if no input is provided', () => {
        const fakeEvent = { preventDefault: () => console.log('preventDefault') };
        const component = shallow(<SearchGifsForm submit={clickFn} />);
        component
            .find('#searchForm')
            .simulate('submit', fakeEvent);
        expect(component.find('InlineError').length).toBe(1)
    });

    it('should submit form', () => {
        let called = false;
        const fakeEvent = { preventDefault: () => called = true };
        const component = shallow(<SearchGifsForm submit={clickFn} />);
        component
            .find('#searchForm')
            .simulate('submit', fakeEvent);
        expect(called).toBe(true);
    });
});

