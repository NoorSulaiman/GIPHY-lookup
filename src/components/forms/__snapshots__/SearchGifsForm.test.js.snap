// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchGifsForm should render input form correctly with given function 1`] = `
<Form
  as="form"
  id="searchForm"
  loading={false}
  onSubmit={[Function]}
>
  <FormGroup
    inline={true}
  >
    <FormField
      error={false}
      id="formField"
    >
      <input
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search for GIFs"
        type="text"
        value=""
      />
    </FormField>
    <Button
      as="button"
      id="searchButton"
      role="button"
    >
      Search
    </Button>
  </FormGroup>
</Form>
`;
